### PR TITLE
Ws 8.3.1 map component search query backend

### PR DESCRIPTION
**Pull request by PrasantiNalla**
created a route path to get nearest 5 locations of whale sightings.

### CODE REVIEW

File: WhaleSpotting/Controllers/TripPlannerController.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    private readonly ITripPlannerService _tripPlannerService;
+
+    public TripPlannerController(ITripPlannerService tripPlannerService)
+    {
+        _tripPlannerService = tripPlannerService;
+    }
+
+    [HttpGet("")]
+    public ActionResult<List<TripPlannerResponse>> GetNearBySightings([FromQuery] TripPlannerRequest input)
```
Comment (francescastocco) [inline WhaleSpotting/Controllers/TripPlannerController.cs:20]: Can we rename this parameter so it is self-explanatory?
```suggestion
    public ActionResult<List<TripPlannerResponse>> GetNearBySightings([FromQuery] TripPlannerRequest tripPlannerRequest)
```

File: WhaleSpotting/Controllers/TripPlannerController.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        try
+        {
+            var response = _tripPlannerService.ListNearBySightings(input.lat, input.lon);
+            return response;
```
Comment (francescastocco) [inline WhaleSpotting/Controllers/TripPlannerController.cs:25]: Can we rename this variable to be more self-explanatory?
```suggestion
            var topFiveSightings = _tripPlannerService.ListNearBySightings(input.lat, input.lon);
            return topFiveSightings;
```

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs [hunk 3]
```diff
@@ -90,5 +93,25 @@ public List<WhaleSightingResponse> ListApprovedSightings()
...
+        try
+        {
+            List<TripPlannerResponse> SighingsList = context.WhaleSightings
+             .Where(ws => (int)ws.ApprovalStatus == 1)
+             .Include(ws => ws.WhaleSpecies)
+             .Select(x => new TripPlannerResponse(x, inputLat, inputLon))
+             .AsEnumerable()
+             .OrderBy(x => x.Distance)
+             .Take(5)
+             .ToList();
```
Comment (francescastocco) [inline WhaleSpotting/Repositories/WhaleSightingRepo.cs:108]: Can you change the indentation on these to 4 spaces?

**Review by francescastocco — CHANGES_REQUESTED** [review]
This looks really good, well done!

File: WhaleSpotting/Controllers/TripPlannerController.cs (hunk 1 above)
Comment (PrasantiNalla) [reply WhaleSpotting/Controllers/TripPlannerController.cs:20]: changed

File: WhaleSpotting/Controllers/TripPlannerController.cs (hunk 2 above)
Comment (PrasantiNalla) [reply WhaleSpotting/Controllers/TripPlannerController.cs:25]: changed

File: WhaleSpotting/Repositories/WhaleSightingRepo.cs (hunk 3 above)
Comment (PrasantiNalla) [reply WhaleSpotting/Repositories/WhaleSightingRepo.cs:108]: changed

**Review by francescastocco — APPROVED** [review]
Nice!

**Pull request merged**